general-superstaq/general_superstaq/superstaq_exceptions.py

```diff
@@ -2,49 +2,30 @@
 
 from typing import Optional
 
-import requests
-
 
 class SuperstaqException(Exception):
     """An exception for errors coming from Superstaq's API."""
 
-    def __init__(self, message: str, status_code: Optional[int] = None):
+    def __init__(self, message: str, status_code: Optional[int] = None) -> None:
         """Initializes the `SupertaqException` class.
 
         Args:
-            status_code: An HTTP status code, if coming from an HTTP response with a failing
-                status.
+            status_code: An HTTP status code, if coming from an HTTP response with a failing status.
             message: A message corresponding to the HTTP response status code. Defaults to None.
         """
-        super().__init__(f"Status code: {status_code}, Message: '{message}'")
         self.status_code = status_code
         self.message = message
 
-
-class SuperstaqModuleNotFoundException(SuperstaqException):
-    """An exception for Superstaq features requiring an uninstalled module."""
-
-    def __init__(self, name: str, context: str):
-        """Initializes the `SuperstaqModuleNotFoundException` class.
-
-        Args:
-            name: The missing module name.
-            context: The context for the exception.
-        """
-        message = f"'{context}' requires module '{name}'"
-        super().__init__(message)
-
-
-class SuperstaqNotFoundException(SuperstaqException):
-    """An exception for errors from Superstaq's API when a resource is not found."""
-
-    def __init__(self, message: str):
-        """Intializes the `SuperstaqNotFoundException` class.
-
-        Args:
-            message: The message to be displayed for this exception.
-        """
-        super().__init__(message, status_code=requests.codes.not_found)
+        if status_code is None:
+            err_msg = f"{message}"
+        else:
+            status_msg = (
+                "400, non-retriable error making request to Superstaq API"
+                if (status_code == 400)
+                else str(status_code)
+            )
+            err_msg = f"{message} (Status code: {status_msg})"
+        super().__init__(err_msg)
```

Comment:
nit:
```suggestion
        if status_code is not None:
            status_msg = (
                "400, non-retriable error making request to Superstaq API"
                if (status_code == 400)
                else str(status_code)
            )
            message = f"{message} (Status code: {status_msg})"
        super().__init__(message)
```

Reply:
also would it make sense to move all of this to `SuperstaqServerException`, and remove the status code from `SuperstaqException` entirely?

Reply:
> also would it make sense to move all of this to SuperstaqServerException, and remove the status code from SuperstaqException entirely?

yes I think this makes sense, will make the changes